src/main/kotlin/org/jetbrains/research/testspark/helpers/ClassTypeProcessHelper.kt

```diff
@@ -0,0 +1,71 @@
+package org.jetbrains.research.testspark.helpers
+
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiMethod
+import com.intellij.psi.PsiModifier
+import com.intellij.psi.util.PsiTreeUtil
+import org.jetbrains.research.testspark.core.data.ClassType
+
+object ClassTypeProcessHelper {
+    /**
```

Comment:
Actually, we tend to always opt for an OOP-like implementation in TestSpark by creating classes and methods, thus increasing the number of entities in the project. We can try to use a Kotlin-like implementation by creating extension methods. 

All the methods of `ClassTypeProcessHelper` may be replaced with extension methods like so:
```kotlin
fun PsiClass.getClassType(): ClassType { ... }

fun PsiClass.getDisplayName(): String { ... }

// ...

myPsiClass.getClassType()
myPsiClass.getDisplayName()
```
Such an approach does not introduce a new class entity, which is generally reasonable.

@arksap2002, @pderakhshanfar, What do you think about that?

Reply:
I agree that using extension methods to eliminate unnecessary classes is a good idea. However, as we expand support for more languages, such an approach might significantly increase the number of lines of code.

In this [PR](https://github.com/JetBrains-Research/TestSpark/pull/223), I introduce a new level of abstraction for handling PsiMethods. The primary motivation is to accommodate the differences in how we interact with PSI structures across various languages. For instance, Kotlin uses KtClass instead of PsiClass [1].

To address these differences, I encapsulate all the PSI logic within wrapper classes (PsiMethodWrapper, PsiClassWrapper). This ensures that methods interacting with PSI structures outside of PsiHelper do not directly deal with KtClass/KtNamedFunction or PsiClass/PsiMethod, but rather with the corresponding wrappers. This abstraction is crucial for maintaining consistency and reducing complexity when supporting multiple languages.

Therefore, certain functions must be included within the PsiHelperInterface and implemented in JavaPsiHelper, KotlinPsiHelper, and other language-specific helpers.

In sum, my recommendation is to firstly merge the new way of parsing and then think how to better and more generic extend it all

[1] https://stackoverflow.com/questions/58113324/why-cannot-i-get-psiclass-from-kotlin-but-successful-in-java#:~:text=PsiClass%20is%20used%20only%20for,Kotlin%20you%20should%20use%20KtClass